src/tests/test_dynamic_api/test_temporal_actions.py

```diff
@@ -6,205 +6,173 @@
+def gebied(gebieden_models, stadsdelen, buurt):
     """
     Creates gebied that is connected to Stadsdeel Zuidoost.
     """
-    Gebied = bagh_models["ggw_gebied"]
+    Gebied = gebieden_models["ggwgebieden"]
     gebied = Gebied.objects.create(
-        id="03630950000019_001",
+        id="03630950000019.1",
         identificatie="03630950000019",
         volgnummer=1,
         registratiedatum=datetime(2015, 1, 1, 5, 40, 12),
         begin_geldigheid=date(2014, 2, 20),
         naam="Bijlmer-Centrum",
-        stadsdeel=bagh_stadsdeelen[1],
+    )
+    Gebied.bestaat_uit_buurten.through.objects.create(
+        ggwgebieden_id="03630950000019.1", bestaat_uit_buurten_id="03630000000078.1"
     )
     return gebied
 
 
 @pytest.fixture()
-def bagh_buurt(bagh_buurt_model, bagh_stadsdeelen, bagh_wijk):
-    return bagh_buurt_model.objects.create(
-        id="03630000000078_001",
+def buurt(gebieden_models, stadsdelen, wijk):
+    Buurt = gebieden_models["buurten"]
+    return Buurt.objects.create(
+        id="03630000000078.1",
         identificatie="03630000000078",
         volgnummer=1,
         code="A00a",
         naam="Kop Zeedijk",
-        stadsdeel=bagh_stadsdeelen[1],
-        wijk=bagh_wijk,
+        ligt_in_wijk=wijk,
     )
 
 
 @pytest.fixture()
-def bagh_wijk(bagh_wijk_model, bagh_stadsdeelen):
-    return bagh_wijk_model.objects.create(
-        id="03630012052022_001",
+def wijk(gebieden_models, stadsdelen):
+    Wijk = gebieden_models["wijken"]
+    return Wijk.objects.create(
+        id="03630012052022.1",
         identificatie="03630012052022",
         volgnummer=1,
+        begin_geldigheid=date(2015, 1, 1),
         code="H36",
         naam="Sloterdijk",
-        stadsdeel=bagh_stadsdeelen[1],
+        ligt_in_stadsdeel=stadsdelen[1],
     )
 
 
 @pytest.mark.django_db
 class TestViews:
-    def test_list_contains_all_objects(
-        self, api_client, filled_router, bagh_schema, bagh_stadsdeelen
-    ):
+    def test_list_contains_all_objects(self, api_client, stadsdelen):
         """ Prove that default API response contains ALL versions."""
-        url = reverse("dynamic_api:bagh-stadsdeel-list")
+        url = reverse("dynamic_api:gebieden-stadsdelen-list")
         response = api_client.get(url)
 
         assert response.status_code == 200, response.data
-        assert len(response.data["_embedded"]["stadsdeel"]) == 2, response.data["_embedded"][
-            "stadsdeel"
+        assert len(response.data["_embedded"]["stadsdelen"]) == 2, response.data["_embedded"][
+            "stadsdelen"
+        ]
+
+    def test_filtered_list_contains_only_correct_objects(self, api_client, stadsdelen, buurt):
+        """Prove that date filter displays only active-on-that-date objects."""
+        url = reverse("dynamic_api:gebieden-stadsdelen-list")
+        response = api_client.get(f"{url}?geldigOp=2015-01-02")
+
+        assert response.status_code == 200, response.data
```

Comment:
By including a message in the assert statement (eg the `response.data` here), you will prevent pytest from rewriting the assert statement **and** inserting introspection details  If that's the intention, fine. If not you should leave out the message.

Reply:
I agree. Actually, I never add messages in pytest asserts. But lots of the tests have those messages. Maybe a good idea to discuss pros/cons in the team. I find the messages are also cluttering the test-code during code-reading.

Reply:
Logic behind current assert and message is simple, if it aint 200 - it should have details in response.data on what went wrong. Saves time when debugging.